ml-algorithms/src/test/java/org/opensearch/ml/engine/encryptor/EncryptorImplTest.java

```diff
@@ -565,6 +564,49 @@ private GetResponse prepareMLConfigResponse(String tenantId) throws IOException
         return new GetResponse(getResult);
     }
 
+    @Test
+    public void encrypt_MasterKeyFieldMismatch_ShouldThrowNPE() throws IOException {
+        // This test simulates the case where the document ID is `master_key_<hash>`
+        // but the actual `_source` only contains `master_key` (as expected in real DDB).
+
+        doAnswer(invocation -> {
+            ActionListener<Boolean> actionListener = (ActionListener) invocation.getArgument(0);
+            actionListener.onResponse(true); // init index success
+            return null;
+        }).when(mlIndicesHandler).initMLConfigIndex(any());
+
+        // Prepare a GetResponse where the _source has ONLY "master_key"
+        Map<String, Object> sourceMap = Map
+            .of(MASTER_KEY, "m+dWmfmnNRiNlOdej/QelEkvMTyH//frS2TBeS2BP4w=", CREATE_TIME_FIELD, Instant.now().toEpochMilli());
```

Comment:
this string is repeated a lot. Would be nice if it were a constant. Would be even nicer if it were randomly generated.

Reply:
I tried random, but then assertion will be hard...

Reply:
Addressed to put that in a constant.

Reply:
I'm fine with not random as long as the string isn't related to any other string being used!